### PR TITLE
RSDK-3335 Add an example tflite modular resource to the C++ SDK

### PR DESCRIPTION
**Pull request by acmorrow**
This PR adds a new tflite mlmodel example modular resource to the C++ SDK. I've validated that I'm able to run the effective detection person detector model with this setup, but not much else. I've attempted to heavily comment the example so that it provides enough guideposts that someone else could probably use it as a template to integrate with another inference engine.

### CODE REVIEW

File: CMakeLists.txt
Comment (acmorrow) [inline CMakeLists.txt:43]: I think this can be backed out. I'd needed it at one point while I was experimenting with different ways to make a soft dependency on tflite, but I don't think the mechanism I settled on requires it.

File: src/viam/api/CMakeLists.txt
Comment (acmorrow) [inline src/viam/api/CMakeLists.txt:141]: Workaround for broken codegen until #113 merges. @zaporter-work had suggested we just merge this if it was going to be a while so that others didn't need to manually apply this in order to be able to work. If #113 merges before this I will remove it.

File: src/viam/examples/modules/CMakeLists.txt
Comment (acmorrow) [inline src/viam/examples/modules/CMakeLists.txt:15]: This can also be backed out now that I'm using `find_path` / `find_library` below.

File: src/viam/examples/modules/CMakeLists.txt
Comment (acmorrow) [inline src/viam/examples/modules/CMakeLists.txt:46]: In theory we could move this to a new `FindTensorflowLite.cmake` in our modules directory, but since I don't know for how long this module will remain in the C++ SDK, that effort didn't seem justified.

File: src/viam/sdk/common/proto_type.hpp
Comment (acmorrow) [inline src/viam/sdk/common/proto_type.hpp:65]: Potentially this could be `decltype(auto)` but in practice I don't think it matters here.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [inline src/viam/examples/modules/mlmodelservice_tflite.cpp:28]: This should have a class level comment that gives an overview and also documents the configuration parameters.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [inline src/viam/examples/modules/mlmodelservice_tflite.cpp:691]: @stuqdog - I'd like some more details on what this field does / means.

File: src/viam/api/CMakeLists.txt
Comment (acmorrow) [reply src/viam/api/CMakeLists.txt:141]: I saw! I have already backed this out locally.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (tessavitabile) [inline src/viam/examples/modules/mlmodelservice_tflite.cpp:70]: If we failed to reconfigure, the resource should behave as if it is closed and not allow requests.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (tessavitabile) [inline src/viam/examples/modules/mlmodelservice_tflite.cpp:55]: Trailing comment

**Review by tessavitabile — COMMENTED** [review]
Looks great! Added a couple comments. No need to wait for my sign-off.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:70]: Great. I wasn't sure what was expected. I think I can make this somewhat simpler if I can allow the service to become closed on a failed reconfiguration.

File: etc/docker/Dockerfile.debian.bookworm
Comment (acmorrow) [inline etc/docker/Dockerfile.debian.bookworm:1]: This is a drive-by: bookworm has been released so I copied the `sid` `Dockerfile` to `bookworm` and updated it as needed.

**Review by acmorrow — COMMENTED** [review]
Updated, please take another look. I cleaned up some comments, added some more, and reworked the reconfiguration logic to put the service into a stopped state on a failed reconfiguration.

File: src/viam/sdk/services/mlmodel/client.cpp
Comment (bhaney) [reply src/viam/sdk/services/mlmodel/client.cpp:51]: I agree that after the flat tensors are added, this will not be necessary- the type check can always be added in later if needed or if flat tensors turn out not to be viable.

**Review by bhaney — COMMENTED** [review]
So far from what I can tell, this looks good and infer() is doing everything right - I'm going to take one more pass at this tomorrow morning now having digested everything and then should be able to give full approval!

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (stuqdog) [inline src/viam/examples/modules/mlmodelservice_tflite.cpp:364]: (nit) s/I'm/we're? Just thinking from the perspective of a third party user expecting an authoritative Viam voice in the example.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (stuqdog) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:741]: Per offline conversation, I think it's an artifact, model/resourceAPI should give all that's necessary.

**Review by stuqdog — APPROVED** [review]
I can't speak super well to the accuracy of some of the mlmodel logic, but generally looks good to me! Left a couple comments about comments, no need to re-request review though.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (bhaney) [inline src/viam/examples/modules/mlmodelservice_tflite.cpp:421]: Should we set a limit to number of threads someone can request? In Go, we warn the user that we limited their threads to maxCPUs/2 if they requested a larger amount.

**Review by bhaney — APPROVED** [review]
Just really one comment about a limit of num_threads - otherwise the meat of the methods all look good to me!

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:3]: I guess I think of them as "system" includes rather than std includes, so I grouped them. I'll fold them in with the standard library includes.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:84]: Fair. At the time I wrote it I wasn't sure it was needed. Changed to `The framework does not offer protection`

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:255]: I forgot a verb there. Changed to:
```
        // Wait for our state to be valid or stopped and then an
        // obtain a shared_ptr to state if valid, incrementing the
        // refcount, or throws if the service is stopped. We don't
        // need to deal with interruption or shutdown because the gRPC
        // layer will drain requests during shutdown, so it shouldn't
        // be possible for callers to get stuck here.
```

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:138]: I've made the check exact and updated the comment.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:364]: I made the comment stronger.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (acmorrow) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:155]: Sure, done.

**Comment by acmorrow** [comment]
@bhaney and @stuqdog - Thanks for the reviews and approvals! I believe I've addressed all open comments, with one standing question about whether we want a ticket for the `ResourceType` change, which would let me clean up that comment with a `TODO(RSDK-ABCD): `.

File: src/viam/examples/modules/mlmodelservice_tflite.cpp
Comment (bhaney) [reply src/viam/examples/modules/mlmodelservice_tflite.cpp:419]: Sounds good to me! I'm good with leaving things as is

**Pull request merged**
